public/app/core/logs_model.ts

```diff
@@ -459,12 +466,13 @@ export function logSeriesToLogsModel(logSeries: DataFrame[]): LogsModel | undefi
   }
 
   const limits = logSeries.filter((series) => series.meta && series.meta.limit);
-  const limitValue = Object.values(
-    limits.reduce((acc: any, elem: any) => {
-      acc[elem.refId] = elem.meta.limit;
-      return acc;
-    }, {})
-  ).reduce((acc: number, elem: any) => (acc += elem), 0) as number;
+  const lastLimitPerRef = limits.reduce<Record<string, number>>((acc, elem) => {
+    acc[elem.refId ?? ''] = elem.meta?.limit ?? 0;
+
+    return acc;
+  }, {});
```

Comment:
This is a slight change in behaviour. The types say that `refId` can potentially be undefined, but that was being ignored. Now, I default it to emptystring. I _think_ this is safe - in practice it means that previously missing-refid queries would be merged with queries with a refId of the string `undefined`, and now they'll just be merged with empty strings.

Reply:
i agree, this is a lot saner behavior.

though, if your goal is no-behavior-change, this could do the job 😁 

```diff
-    acc[elem.refId ?? ''] = elem.meta?.limit ?? 0;
+    acc[elem.refId ?? 'undefined'] = elem.meta?.limit ?? 0;
```

Reply:
i agree, this is a lot saner behavior.

though, if your goal is no-behavior-change, this could do the job 😁 

```diff
-    acc[elem.refId ?? ''] = elem.meta?.limit ?? 0;
+    acc[elem.refId ?? 'undefined'] = elem.meta?.limit ?? 0;
```